src/main/java/com/yugabyte/sample/apps/SqlInserts.java

```diff
@@ -105,6 +117,16 @@ private PreparedStatement getPreparedSelect() throws Exception {
     }
     return preparedSelect;
   }
+  
+  // Returns false if the sleep is interrupted
+  static boolean sleep(int durationMs) {
```

Comment:
I don't think just by making threads to sleep and retry will solve the problem reported by @kmuthukk . 

We'll need to find the next healthy connection that is available in the cluster, we are not trying to solve running sample-apps against an unhealthy cluster.

Reply:
AppBase.getPostgresConnection() would attempt different contact points in a loop until a good contact point is found.